### PR TITLE
Migrate `run-make/rustdoc-output-path` to rmake

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/rustdoc-output-path/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Checks that if the output folder doesn't exist, rustdoc will create it.
+
+use run_make_support::{rustdoc, tmp_dir};
+
+fn main() {
+    rustdoc().input("foo.rs").output(tmp_dir().join("foo/bar/doc")).run();
```
Comment (jieyouxu) [inline tests/run-make/rustdoc-output-path/rmake.rs:6]: I know this is a direct translation of the original test, but the original test doesn't actually check if the path exists -- it only checked that `rustdoc` with non-existent output folder returns a success status code.

Comment (GuillaumeGomez) [reply tests/run-make/rustdoc-output-path/rmake.rs:6]: Should I add this check then? Doesn't hurt to check that too I suppose.

**Comment by jieyouxu** [comment]
Feel free to r=me after CI is green

**Comment by GuillaumeGomez** [comment]
Rebased.

@bors r=jieyouxu rollup

**Pull request merged**